app/Http/Controllers/BeatmapsController.php

```diff
@@ -481,16 +481,20 @@ public function userScore($beatmapId, $userId)
      * @urlParam user integer required Id of the [User](#user).
      *
      * @queryParam legacy_only integer Whether or not to exclude lazer scores. Defaults to 0. Example: 0
-     * @queryParam mode The [Ruleset](#ruleset) to get scores for. Defaults to beatmap mode
+     * @queryParam mode (deprecated) The [Ruleset](#ruleset) to get scores for. Defaults to beatmap ruleset. No-example
+     * @queryParam ruleset The [Ruleset](#ruleset) to get scores for. Defaults to beatmap ruleset. Example: osu
      */
     public function userScoreAll($beatmapId, $userId)
     {
         $beatmap = Beatmap::scoreable()->findOrFail($beatmapId);
-        $mode = presence(get_string(request('mode'))) ?? $beatmap->mode;
+        $ruleset = presence(get_string(request('ruleset'))) ?? presence(get_string(request('mode')));
```

Comment:
this is going to `Arr::get` twice, isn't it? 🤔

Reply:
it will, but eh 🤷‍♂️